### PR TITLE
fix(install-local.sh): do more repair to `FARCH` array

### PR DESCRIPTION
**Pull request by oklopfer**
## Purpose
`FARCH` only worked if the given pacscript's `arch` array was singular (i.e. if more than one foreign architecture was listed, then it did not work)

## Approach
Fix the check to loop properly

## Progress

- [x] Did it
- [x] Tested

## Checklist

- [x] I confirm that I have read the [contributing guidelines](https://github.com/pacstall/pacstall/blob/develop/CONTRIBUTING.md), and this pull request is abiding by all the clauses stated in the guideline.

### CODE REVIEW

File: misc/scripts/install-local.sh [hunk 1]
```diff
@@ -252,23 +252,25 @@ function get_incompatible_releases() {
...
+        for pacarch in "${input[@]}"; do
+            for farch in "${FARCH[@]}"; do
+                if [[ "${pacarch}" == "${farch}" ]]; then
+                    fancy_message warn "This package is for ${BBlue}${farch}${NC}, which is a foreign architecture"
+                    ret=0
+                fi
+            done
+        done
+    fi
+    if ((ret==1)); then
```
Comment (Elsie19) [inline misc/scripts/install-local.sh:270]:
```suggestion
    if ! ((ret)); then
```

**Review by Elsie19 — CHANGES_REQUESTED** [review]

File: misc/scripts/install-local.sh (hunk 1 above)
Comment (oklopfer) [reply misc/scripts/install-local.sh:270]: Doesn't work.

Comment (oklopfer) [reply misc/scripts/install-local.sh:270]: with mine:
```bash
[~] rhino-live@rhino-live $ pacstall -I aws-cli-v2-bin
(aws-cli-v2-bin) Do you want to view/edit the pacscript? [y/N] n
[+] INFO: Sourcing pacscript
[!] ERROR: This Pacscript does not work on arm64
[~] rhino-live@rhino-live $ 
```
with yours:
```bash
[~] rhino-live@rhino-live $ pacstall -I aws-cli-v2-bin
(aws-cli-v2-bin) Do you want to view/edit the pacscript? [y/N] n
[+] INFO: Sourcing pacscript
[~] rhino-live@rhino-live $ 
```

File: misc/scripts/install-local.sh [hunk 2]
```diff
@@ -252,23 +252,29 @@ function get_incompatible_releases() {
 }
 
 function is_compatible_arch() {
-    local input=("${@}")
+    local input=("${@}") ret=1
```
Comment (Elsie19) [inline misc/scripts/install-local.sh:255]:
```suggestion
    local input=("${@}") ret=1 pacarch farch
```

**Review by Elsie19 — CHANGES_REQUESTED** [review]

File: misc/scripts/install-local.sh (hunk 2 above)
Comment (oklopfer) [reply misc/scripts/install-local.sh:255]: Why would I localize those? They’re fill-ins for a for loop.

**Pull request merged**